### PR TITLE
stop placing water in the nether

### PR DESCRIPTION
**Pull request by Awakened-Redstone**

### CODE REVIEW

**Review by StellarWitch7 — APPROVED** [review]
Looks good. Tested it myself. Didn't notice any issues.

File: src/main/java/dev/enjarai/trickster/data/conversion/BlockConversionLoader.java [hunk 1]
```diff
@@ -99,6 +102,17 @@ public boolean convert(Block block, World world, BlockPos pos) {
             world.syncWorldEvent(WorldEvents.BLOCK_BROKEN, pos, Block.getRawIdFromState(blockState));
         }
 
+        if (world.getDimension().ultrawarm()) {
+            if (blockState.isOf(Blocks.WATER)) {
+                blockState = Blocks.AIR.getDefaultState();
+            } else if (blockState.getFluidState().isOf(Fluids.WATER)) {
+                Optional<Boolean> perhapsWaterlogged = blockState.getOrEmpty(Properties.WATERLOGGED);
+                if (perhapsWaterlogged.isPresent() && perhapsWaterlogged.get()) {
+                    blockState.with(Properties.WATERLOGGED, false);
```
Comment (enjarai) [inline src/main/java/dev/enjarai/trickster/data/conversion/BlockConversionLoader.java:111]: What exactly does this line do? BlockStates are immutable, so this wont update the `blockState` variable right?

Comment (Awakened-Redstone) [reply src/main/java/dev/enjarai/trickster/data/conversion/BlockConversionLoader.java:111]: oh, oops

**Comment by StellarWitch7** [comment]
Since the issue @enjarai noticed is fixed, I'll merge this now.

**Pull request merged**